corehq/apps/users/models.py

```diff
@@ -564,9 +564,11 @@ def add_as_web_user(self, domain, role, primary_location_id=None,
             if primary_location_id:
                 self.set_location(domain, primary_location_id, commit=False)
             self.reset_locations(domain, assigned_location_ids, commit=False)
+        user_data = self.get_user_data(domain_obj.name)
         if domain_has_privilege(domain_obj.name, privileges.APP_USER_PROFILES) and profile:
-            user_data = self.get_user_data(domain_obj.name)
```

Comment:
also here

Reply:
This does still require the domain privilege check for profiles and I believe it does do the check. Or maybe i'm misunderstanding your comment